spec/plsql/procedure_spec.rb

```diff
@@ -56,83 +57,59 @@
     before(:all) do
       plsql.connect! CONNECTION_PARAMS
       plsql.execute <<-SQL
         CREATE OR REPLACE FUNCTION test_sum
-          ( p_num1 NUMBER, p_num2 NUMBER )
-          RETURN NUMBER
+          ( p_num1 #{ora_data_type}, p_num2 #{ora_data_type} )
+          RETURN #{ora_data_type}
         IS
         BEGIN
           RETURN p_num1 + p_num2;
         END test_sum;
       SQL
-      plsql.execute <<-SQL
-        CREATE OR REPLACE FUNCTION test_number_1
-          ( p_num NUMBER )
-          RETURN VARCHAR2
-        IS
-        BEGIN
-          IF p_num = 1 THEN
-            RETURN 'Y';
-          ELSIF p_num = 0 THEN
-            RETURN 'N';
-          ELSIF p_num IS NULL THEN
-            RETURN NULL;
-          ELSE
-            RETURN 'UNKNOWN';
-          END IF;
-        END test_number_1;
-      SQL
-      plsql.execute <<-SQL
-        CREATE OR REPLACE PROCEDURE test_integers
-          ( p_pls_int PLS_INTEGER, p_bin_int BINARY_INTEGER, x_pls_int OUT PLS_INTEGER, x_bin_int OUT BINARY_INTEGER )
-        IS
-        BEGIN
-          x_pls_int := p_pls_int;
-          x_bin_int := p_bin_int;
-        END;
-      SQL
     end
   
     after(:all) do
       plsql.execute "DROP FUNCTION test_sum"
-      plsql.execute "DROP FUNCTION test_number_1"
-      plsql.execute "DROP PROCEDURE test_integers"
       plsql.logoff
     end
   
-    it "should process integer parameters" do
-      expect(plsql.test_sum(123,456)).to eq(579)
-    end
-
-    it "should process big integer parameters" do
-      expect(plsql.test_sum(123123123123,456456456456)).to eq(579579579579)
+    it "should get #{ora_data_type} variable type mapped to #{class_.to_s}" do
+      expect( plsql.test_sum(num1, num2) ).to be_a class_
     end
-
-    it "should process float parameters and return BigDecimal" do
-      expect(plsql.test_sum(123.123,456.456)).to eq(BigDecimal("579.579"))
-    end
-
-    it "should process BigDecimal parameters and return BigDecimal" do
-      expect(plsql.test_sum(:p_num1 => BigDecimal("123.123"), :p_num2 => BigDecimal("456.456"))).to eq(BigDecimal("579.579"))
+    it "should process input parameters and return correct result" do
+      expect( plsql.test_sum(num1, num2) ).to eq(expected)
     end
 
     it "should process nil parameter as NULL" do
-      expect(plsql.test_sum(123,nil)).to be_nil
-    end
+      expect(plsql.test_sum(num1,nil)).to be_nil
+    end unless mandatory
 
-    it "should convert true value to 1 for NUMBER parameter" do
-      expect(plsql.test_number_1(true)).to eq('Y')
-    end
-
-    it "should convert false value to 0 for NUMBER parameter" do
-      expect(plsql.test_number_1(false)).to eq('N')
-    end
+  end
 
-    it "should process binary integer parameters" do
-      expect(plsql.test_integers(123, 456)).to eq({:x_pls_int => 123, :x_bin_int => 456})
+  [
+      {:ora_data_type => 'INTEGER',       :class => Bignum,     :num1 => 123456789012,:num2 => 123456789012,:expected => 246913578024 },
```

Comment:
In Jruby 246913578024 appears to be `Fixnum`, not `Bignum` (for Bignum - 20+ digit number seems to be needed).